src/service/owners-impl.js

```diff
@@ -15,20 +15,48 @@
  */
 
 /* eslint-disable no-unused-vars */
+import {Resource, ResourceState} from './resource';
+import {dev, devAssert} from '../log';
+import {isArray} from '../types';
+import {registerServiceBuilderForDoc} from '../service';
+import {Services} from '../services';
+
+const TAG_ = 'Owners';
+
+/**
+ * @param {!Element|!Array<!Element>} elements
+ * @return {!Array<!Element>}
+ */
+function elements_(elements) {
+  return /** @type {!Array<!Element>} */ (isArray(elements)
+    ? elements
+    : [elements]);
+}
+
 /**
  * TODO(lannka): migrate all methods from OwnersDef to here.
  *
  * @interface
  */
 export class Owners {
+  /**
+   * @param {!./ampdoc-impl.AmpDoc} ampdoc
+   */
+  constructor(ampdoc) {
+    /** @const @private {!./resources-impl.ResourcesDef} */
+    this.resources_ = Services.resourcesForDoc(ampdoc);
+  }
+
   /**
    * Assigns an owner for the specified element. This means that the resources
    * within this element will be managed by the owner and not Resources manager.
    * @param {!Element} element
    * @param {!AmpElement} owner
    * @package
    */
-  setOwner(element, owner) {}
+  static setOwner(element, owner) {
```

Comment:
this will introduce a direct file dependency, and likely pulling all the bytes to extensions.